src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -80,17 +80,24 @@ protected override async Task<SearchResult> SearchInternalAsync(SearchOptions se
                     // Begin a transaction so we can perform two atomic reads.
                     using (var transaction = connection.BeginTransaction(IsolationLevel.ReadCommitted))
                     {
-                        searchResult = await SearchImpl(searchOptions, false, connection, cancellationToken, transaction);
+                        try
+                        {
+                            searchResult = await SearchImpl(searchOptions, false, connection, cancellationToken, transaction);
```

Comment:
This should be outside of the try block. If it throws, you don't want to revert.

Reply:
If this throws, searchOptions.CountOnly will already be set to false (due to the if statement on line 78). 

I agree that this doesn't necessarily need to be in the try block and that it's a bit redundant to reset a false value to false a second time. Given this is already merged, how would you recommend moving forward?

Reply:
Not a problem, I would just address it in the next change.